lib/AST/Decl.cpp

```diff
@@ -3383,9 +3383,11 @@ TypeLoc &AssociatedTypeDecl::getDefaultDefinitionLoc() {
 
 SourceRange AssociatedTypeDecl::getSourceRange() const {
   SourceLoc endLoc;
-  if (auto TWC = getTrailingWhereClause())
+  if (auto TWC = getTrailingWhereClause()) {
     endLoc = TWC->getSourceRange().End;
-  else if (!getInherited().empty()) {
+  } else if (!getDefaultDefinitionLoc().isNull()) {
```

Comment:
`if (getDefaultDefinitionLoc().hasLocation())` is preferable here.

Reply:
OK - done.